src/app/inventory/store/masterwork.ts

```diff
@@ -129,25 +129,20 @@ function buildMasterworkInfo(
   defs: D2ManifestDefinitions
 ): DimMasterwork | null {
   const socket = sockets.sockets.find((socket) => Boolean(socket.plug?.plugObjectives.length));
-  if (
-    !socket ||
-    !socket.plug ||
-    !socket.plug.plugObjectives ||
-    !socket.plug.plugObjectives.length
-  ) {
+  if (!socket?.plug?.plugObjectives?.length) {
```

Comment:
I do not know if this is a valid change but I believe it is.

Reply:
i welcome this

Reply:
yup that works